### PR TITLE
Process preprocessing tasks in pachw [PFS-65]

### PR DESCRIPTION
**Pull request by brycemcanally**
This PR moves the processing of preprocessing tasks (i.e. creating / diffing datums) to pachw. This allows users to scale job preprocessing without scaling pipeline workers, which often use more expensive resources such as GPUs. The code for preprocessing and processing has been more explicitly split in the code, but the preprocessing code still lives in the worker package to enable fallback on the old model without introducing cyclic dependencies (i.e. worker -> pps). Eventually we will remove the fallback and move the preprocessing code to a pps package.

### CODE REVIEW

**Comment by FahadBSyed** [comment]
Looks like there is an import cycle in one of the tests:
```
2024-01-04T15:46:56 src/server/worker/pipeline/transform/cache.go:1: : import cycle not allowed in test (typecheck)
2024-01-04T15:46:56 package transform
```

**Review by FahadBSyed — CHANGES_REQUESTED** [review]
Code changes look good barring the import cycle. The explicit renaming makes the code very readable and the choice to instantiate the server with a pach client is very straight forward.

File: etc/helm/pachyderm/values.yaml [hunk 1]
```diff
@@ -690,6 +690,9 @@ pachw:
   # when enabled, pachw instances can still run in their own dedicated kubernetes deployment if maxReplicas is greater than 0.
   # For more control of where pachw instances run, 'inSidecars' should be disabled.
   inSidecars: false
+  # When set to true, preprocessing tasks are only processed by pachw instances.
+  # When set to false, pipeline workers will also participate in processing preprocessing tasks.
+  onlyPreprocessing: true
```
Comment (FahadBSyed) [inline etc/helm/pachyderm/values.yaml:695]: The name of this flag may lead to confusion (as well as `PACHW_ONLY_PREPROCESSING`). Without the comment, I'd interpret this as pachw only processes preprocessing tasks and no other kinds of tasks.
That said, I'm not sure what a better name for this is. Maybe `handleAllPreprocessingTasks`?

Comment (brycemcanally) [reply etc/helm/pachyderm/values.yaml:695]: Yeah, I was actually going to bring this up in standup. I don't really like the name either. I'll wait until then when we can discuss it before merging.

**Review by FahadBSyed — APPROVED** [review]
I left one non-blocking comment, otherwise the changes look good to me.

**Review by emmajsadams — APPROVED** [review]
I am primarily reviewing the `python-sdk/pachyderm_sdk/api/pfs/__init__.py` changes. As you mentioned in Slack they come from https://github.com/pachyderm/pachyderm/pull/9633 and are autogenerated so they seem safe to me!

**Pull request merged**